text/3371-cargo-target-directories.md

```diff
@@ -221,6 +227,32 @@ bringing the registry cache in their build directory for example.
 I feel this require an hard-to-break naming scheme, which I don't have the skills nor motivation to design. Instead, I prefer explicitely telling the naming scheme is not to
 be considered stable and allow more invested people to experiment with the feature and find something solid.
 
+## Make the naming conflict less easily
+
+The (possibly) conflicting naming scheme used here is something that can easily be fixed: instead of just `/cargo-caches/work-project`, use something like `/cargo-caches/work-project-<blake3 hash of full path to 'work-project' directory>`, like `targo` does. This approach has at least two defaults and one advantage:
+
+- **Advantage**: conflicts are pretty much impossible while the naming scheme is still predictable
+- **Disadvantage**: external tools now have to know about the naming scheme internal more than just "set `$CARGO_TARGET_DIRECTORIES` and append crate directory name" and changing the hash method or length could heavily break them (even with us specifying it as unstable, we all know how that goes in real life). We would also have to specify when is the hash resolved (before symlink resolution or after).
```

Comment:
I don't think this is an issue as external tools should ask [`cargo_metadata` for the `target_directory`](https://docs.rs/cargo_metadata/latest/cargo_metadata/struct.Metadata.html#structfield.target_directory) rather than re-invent the lookup logic.

Reply:
I feel very dumb because I think I made this exact argument somewhere else. Thanks for catching it !